pkg/util/steps/action.go

```diff
@@ -17,17 +17,32 @@ type actionFunction func(context.Context) error
 
 // Action returns a Step which will execute the action function `f`. Errors from
 // `f` are returned directly.
-func Action(f actionFunction) Step {
-	return actionStep{f}
+// The metricsTopic parameter is optional and only the first element is considered if present.
+func Action(f actionFunction, metricsTopic ...string) Step {
```

Comment:
Why is `metricsTopic` a variadic parameter?  Is it to allow for zero to be specified (i.e. an optional parameter?

I would prefer to take the function and derive a name from said function rather than hardcoding strings above.  Something similar to the `friendlyName` function within steps.  Then we don't have to worry about passing in any data as it's already there within the function.

Reply:
@bennerv Thanks for the comment! Yes, `metricsTopic` is an optional param here, we just need to pass it wherever we need to assign a metrics topic to the action. For the conception of `friendlyName`, that is actually my first solution, but seems it is not working properly in some scenario, I got nil pointer issues when running unit test for the `runSteps` func. The idea of passing `metricsTopic` is based on the suggestions from Amber in slack channel: https://coreos.slack.com/archives/C02ULBRS68M/p1656557874242279?thread_ts=1656556973.718739&cid=C02ULBRS68M